front/src/routes/integration/all/zwavejs-ui/ZwaveJSUIDeviceBox.jsx

```diff
@@ -0,0 +1,214 @@
+import { Component } from 'preact';
+import { Text, Localizer, MarkupText } from 'preact-i18n';
+import cx from 'classnames';
+import get from 'get-value';
+
+import DeviceFeatures from '../../../../components/device/view/DeviceFeatures';
+
+import { connect } from 'unistore/preact';
+
+class ZwaveJSUIDeviceBox extends Component {
+  componentWillMount() {
+    this.setState({
+      device: this.props.device
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      device: nextProps.device
+    });
+  }
+
+  updateName = e => {
+    this.setState({
+      device: {
+        ...this.state.device,
+        name: e.target.value
+      }
+    });
+  };
+
+  updateRoom = e => {
+    this.setState({
+      device: {
+        ...this.state.device,
+        room_id: e.target.value
+      }
+    });
+  };
+
+  saveDevice = async () => {
+    this.setState({
+      loading: true,
+      errorMessage: null
+    });
+    try {
+      let deviceDidNotExist = this.state.device.id === undefined;
+      const savedDevice = await this.props.httpClient.post(`/api/v1/device`, this.state.device);
+      if (deviceDidNotExist) {
+        savedDevice.alreadyExist = true;
+      }
+      this.setState({
+        device: savedDevice
+      });
+    } catch (e) {
+      let errorMessage = 'integration.zwavejs-ui.error.defaultError';
+      if (e.response.status === 409) {
+        errorMessage = 'integration.zwavejs-ui.error.conflictError';
+      }
+      this.setState({
+        errorMessage
+      });
+    }
+    this.setState({
+      loading: false
+    });
+  };
+
+  deleteDevice = async () => {
+    this.setState({
+      loading: true,
+      errorMessage: null,
+      tooMuchStatesError: false,
+      statesNumber: undefined
+    });
+    try {
+      if (this.state.device.created_at) {
+        await this.props.httpClient.delete(`/api/v1/device/${this.state.device.selector}`);
+      }
+      this.props.getZwaveJSUIDevices();
```

Comment:
Don't you want to only remove deleted device from list instead of reload all?

Reply:
Good enough for this MVP imo !